docs/next/guides/upgrade-and-migration/release-notes.md

```diff
@@ -15,6 +15,36 @@ tags:
 
 [[toc]]
 
+## 12.1.0
+
+Released on 28th of June, 2022
+
+**Added**
+- Added smart positioning of comment boxes. [#4785](https://github.com/handsontable/handsontable/issues/4785)
+- Added smoother row and column moving when some row or columns are hidden. [#7613](https://github.com/handsontable/handsontable/issues/7613)
+- Added [`getNearestNotHiddenIndex()`](@/api/indexMapper.md#getnearestnothiddenindex), a new method that finds the visual index of the nearest not-hidden row or column and works even with large numbers of hidden rows or columns. The previous method, [`getFirstNotHiddenIndex()`](@/api/indexMapper.md#getfirstnothiddenindex), still works, but is marked as deprecated. [#9064](https://github.com/handsontable/handsontable/issues/9064)
+- Added a Czech translation. [#9343](https://github.com/handsontable/handsontable/issues/9343)
+- Added a Serbian translation. [#9469](https://github.com/handsontable/handsontable/issues/9469)
+- Added new hooks: [`beforeColumnFreeze`](@/api/hooks.md#beforecolumnfreeze), [`afterColumnFreeze`](@/api/hooks.md#aftercolumnfreeze), [`beforeColumnUnfreeze`](@/api/hooks.md#beforecolumnunfreeze), and [`afterColumnUnfreeze`](@/api/hooks.md#aftercolumnunfreeze). [#9248](https://github.com/handsontable/handsontable/issues/9248)
```

Comment:
This is a comment to the file `api/hooks.md`, but I don't have an option to comment in that file, so I am doing it here:

```
| freezePerformed | `boolean` | If `true`: the column is going to freeze. If `false`: the column is not going to freeze. |
```

In the above description in `beforeColumnFreeze`, it is not clear why would a column not freeze. I propose to rewrite it to the following (discussed with @wszymanski):

```
| freezePerformed | `boolean` | If `true`: the column is going to freeze. If `false`: the column is not going to freeze (which might happen if the column is already frozen). |
```

The same applies to the description in `beforeColumnUnfreeze`:

```
| unfreezePerformed | `boolean` | If `true`: the column is going to unfreeze. If `false`: the column is not going to unfreeze. |
```

Could be changed to:

```
| unfreezePerformed | `boolean` | If `true`: the column is going to unfreeze. If `false`: the column is not going to unfreeze (which might happen if the column is already unfrozen). |
```

Reply:
@warpech - done: [df795d5](https://github.com/handsontable/handsontable/pull/9599/commits/df795d58f2aa3c8abc2e8e6195dfaa9ad34ec5a1)